### DISCUSSION TITLE
Removing GeoJSON layer from memory

### DISCUSSION THREAD

**Discussion by brad28b** [Leaflet/Leaflet#9164] [Q&A]
My application loads various GeoJSON layers. The end-user can disable or enable them via a menu. I am finding that when I remove a layer, it is not released from memory. And adding them again increases the size of the Java VM Heap each time. After a period of time enabling and disabling various layers, eventually it crashes the browser. Tried all versions from 1.0.0 to 1.9.4 and it exists in all versions.

I'm adding the GeoJSON layer like this:

```
datalayer1 = L.geoJSON(xhr.response, {
    onEachFeature: onEachFeature,
    style: myStyle
}).addTo(map);
```

And removing it like this:

```
datalayer1.remove();
map.removeLayer(datalayer1);
datalayer1=null;
```

Any clues on how to release from memory?

**Comment by jonkoops** [comment]
If you can reproduce this reliably, please log a bug with steps on how to reproduce and diagnose the issue.
